File: webapp_snapshot/fixCache.py
import re

filepath = 'src/components/ProductTreeSelector.tsx'
with open(filepath, 'r', encoding='utf-8') as f:
    content = f.read()

# Add cache busting to the fetch
content = content.replace("fetch('/api/catalogs')", "fetch(`/api/catalogs?_t=${Date.now()}`)")

with open(filepath, 'w', encoding='utf-8') as f:
    f.write(content)
print("Added cache busting to ProductTreeSelector")

File: webapp_snapshot/fix_cache.js
const fs = require('fs');

function fixFetch(file) {
  let code = fs.readFileSync(file, 'utf8');
  code = code.replace(/await fetch\('\/api\/gastos\?grupo=IVA'\)/, "await fetch('/api/gastos?grupo=IVA', { cache: 'no-store' })");
  code = code.replace(/await fetch\('\/api\/gastos'\)/, "await fetch('/api/gastos', { cache: 'no-store' })");
  fs.writeFileSync(file, code, 'utf8');
}

fixFetch('src/app/cristina-admin/iva/page.tsx');
fixFetch('src/app/cristina-admin/gastos/page.tsx');
